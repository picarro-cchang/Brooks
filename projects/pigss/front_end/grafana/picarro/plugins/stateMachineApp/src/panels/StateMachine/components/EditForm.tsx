import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div className="row">
        <label className="edit-label"> Bank {num}</label>
        {/* <input
          name={"bank" + num}
          className="col-sm-6 edit-input"
          type="text"
          value={this.props.plan.bank_names[num].name}
          onChange={event =>
            this.props.handleBankChange(
              event.target.value,
              num.toString()
            )
          }
          maxLength={14}
        /> */}
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={8}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
