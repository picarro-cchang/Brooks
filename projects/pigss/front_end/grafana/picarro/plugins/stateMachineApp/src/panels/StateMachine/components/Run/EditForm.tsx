import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div key={"bank" + num} className="row">
            <label className="edit-label"> Bank {num}</label>
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div key={portNumber} className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={15}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
