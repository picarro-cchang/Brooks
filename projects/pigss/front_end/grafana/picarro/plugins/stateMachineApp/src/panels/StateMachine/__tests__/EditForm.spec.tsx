import React from "react";
import { shallow, mount } from "enzyme";
import "jest-styled-components";
import EditForm from "../components/EditForm";
import mockPlan from "./../api/__mocks__/mockPlan.json";

const mockClick = jest.fn(element => {
  return element;
});
const mockHandleBankChange = jest.fn();
const mockHandleChannelNameChange = jest.fn();

const defaultProps = {
  uistatus: {
    bank: {
      "1": "READY",
      "2": "READY",
      "3": "DISABLED"
    }
  },
  ws_sender: mockClick,
  plan: mockPlan,
  handleBankChange: mockHandleBankChange,
  handleChannelNameChange: mockHandleChannelNameChange
};

describe("<EditForm />", () => {
  const part = mount(<EditForm {...defaultProps} />);
  const wrapper = shallow(<EditForm {...defaultProps} />);
  const event = {
    target: {
      name: "bank num",
      value: "Bank Number Name"
    }
  };
  const event2 = {
    target: {
      name: "channel num",
      value: "Channel Number Name"
    }
  };

  it("Renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders with correct bank/channel Names", () => {
    const value = wrapper.find('input[name="bank1"]').props().value;
    const chn2value = wrapper.find('input[name="bank12"]').props().value;

    expect(value).toEqual("Bank 1");
    expect(chn2value).toEqual("Ch. 2");
  });

  it("onChange event for Bank", () => {
    wrapper.find('input[name="bank1"]').simulate("change", event);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleBankChange.mock.calls[0]).toEqual([
      "Bank Number Name",
      "1"
    ]);
  });

  it("onChange event for Channel 1", () => {
    wrapper.find('input[name="bank11"]').simulate("change", event2);
    expect(mockHandleChannelNameChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      1
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 2", () => {
    wrapper.find('input[name="bank12"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      2
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 3", () => {
    wrapper.find('input[name="bank13"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      3
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 4", () => {
    wrapper.find('input[name="bank14"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      4
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 5", () => {
    wrapper.find('input[name="bank15"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      5
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 6", () => {
    wrapper.find('input[name="bank16"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      6
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 7", () => {
    wrapper.find('input[name="bank17"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      7
    ]);
    mockHandleChannelNameChange.mockClear();
  });
  it("onChange event for Channel 8", () => {
    wrapper.find('input[name="bank18"]').simulate("change", event2);
    expect(mockHandleBankChange).toHaveBeenCalled();
    expect(mockHandleChannelNameChange.mock.calls[0]).toEqual([
      "Channel Number Name",
      "1",
      8
    ]);
    mockHandleChannelNameChange.mockClear();
  });
});
