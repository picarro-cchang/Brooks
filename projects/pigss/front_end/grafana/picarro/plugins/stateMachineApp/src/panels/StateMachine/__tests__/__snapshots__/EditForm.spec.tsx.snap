// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditForm /> Renders correctly 1`] = `
<div
  className="panel-plan-inner"
>
  <div
    className="gf-form-group"
  >
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Bank 
        1
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={14}
        name="bank1"
        onChange={[Function]}
        type="text"
        value="Bank 1"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 1 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank11"
        onChange={[Function]}
        type="text"
        value="Ch. 1"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 2 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank12"
        onChange={[Function]}
        type="text"
        value="Ch. 2"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 3 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank13"
        onChange={[Function]}
        type="text"
        value="Ch. 3"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 4 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank14"
        onChange={[Function]}
        type="text"
        value="Ch. 4"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 5 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank15"
        onChange={[Function]}
        type="text"
        value="Ch. 5"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 6 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank16"
        onChange={[Function]}
        type="text"
        value="Ch. 6"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 7 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank17"
        onChange={[Function]}
        type="text"
        value="Ch. 7"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 8 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank18"
        onChange={[Function]}
        type="text"
        value="Ch. 8"
      />
    </div>
  </div>
  <div
    className="gf-form-group"
  >
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Bank 
        2
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={14}
        name="bank2"
        onChange={[Function]}
        type="text"
        value="Bank 2"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 1 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank21"
        onChange={[Function]}
        type="text"
        value="Ch. 1"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 2 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank22"
        onChange={[Function]}
        type="text"
        value="Ch. 2"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 3 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank23"
        onChange={[Function]}
        type="text"
        value="Ch. 3"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 4 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank24"
        onChange={[Function]}
        type="text"
        value="Ch. 4"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 5 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank25"
        onChange={[Function]}
        type="text"
        value="Ch. 5"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 6 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank26"
        onChange={[Function]}
        type="text"
        value="Ch. 6"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 7 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank27"
        onChange={[Function]}
        type="text"
        value="Ch. 7"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 8 
      </label>
      <input
        className="col-sm-6 edit-input"
        maxLength={8}
        name="bank28"
        onChange={[Function]}
        type="text"
        value="Ch. 8"
      />
    </div>
  </div>
</div>
`;
